server/routes/bookVisit/selectVisitorsController.ts

```diff
@@ -35,14 +41,15 @@ export default class SelectVisitorsController {
   public submit(): RequestHandler {
     return async (req, res) => {
       const errors = validationResult(req)
+
       if (!errors.isEmpty()) {
         req.flash('errors', errors.array())
-        req.flash('formValues', req.body)
+        req.flash('formValues', matchedData(req, { onlyValidData: false }))
```

Comment:
Makes sure that only data from expected form fields will end up getting saved with `req.flash()` - i.e. not trusting what's in the request.

(The `onlyValidData: true` property is slightly confusing: it means that it won't return an _unexpected_ field, but would return one that had a legitimate field but with _invalid_ data - e.g. it was too short, for example.)